.github/workflows/ci.yml

```diff
@@ -0,0 +1,100 @@
+name: CI
+
+on:
+  pull_request:
+
+env:
+  JAVA_TOOL_OPTIONS: -Xmx4096m
+  GRADLE_OPTS: -Dorg.gradle.daemon=false -Dorg.gradle.parallel=true -Dorg.gradle.workers.max=4 -Xmx4096m
+
+jobs:
+  build:
+    # 4 cpu, 16G ram
+    runs-on: ubuntu-24.04
+    environment: dev
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Prepare
+        uses: ./.github/actions/prepare
+
+      - name: Build and unit test
+        run: |
+          ./gradlew build
+
+      - name: Integration Test
+        run: |
+          ./gradlew integrationTest
+
+      - name: Test results and report
+        uses: ./.github/actions/testResultsReports
+        with:
+          suiteName: 'unitIntegrationTests'
+
```

Comment:
we can possibly also use https://github.com/marketplace/actions/test-summary-action for a nice summary of all the tests generated.